### PR TITLE
Fix server exception constructor throwing exception if the provided json object has no message

### PR DESCRIPTION
**Pull request by tung2744**

### CODE REVIEW

File: sdk/src/main/java/com/oursky/authgear/ServerException.kt [hunk 1]
```diff
@@ -9,9 +9,9 @@ class ServerException(
     val info: JSONObject? = null
 ) : AuthgearException(message) {
     constructor(json: JSONObject) : this(
-        json.getString("name"),
-        json.getString("reason"),
-        json.getString("message"),
+        json.optString("name"),
+        json.optString("reason"),
+        json.optString("message"),
```
Comment (louischan-oursky) [inline sdk/src/main/java/com/oursky/authgear/ServerException.kt:14]: But the constructor accepts a non-nullable message. What is the reason behind this change?

Comment (tung2744) [reply sdk/src/main/java/com/oursky/authgear/ServerException.kt:14]: Is message something must be non-null from the server?
In latte, it is possible to have authgear error without message from custom ui.

Comment (louischan-oursky) [reply sdk/src/main/java/com/oursky/authgear/ServerException.kt:14]: > Is message something must be non-null from the server?

Yes.

> In latte, it is possible to have authgear error without message from custom ui.

For instance?

Comment (tung2744) [reply sdk/src/main/java/com/oursky/authgear/ServerException.kt:14]: Then it seems a bug in custom ui that it always didn't have message now. Maybe I fix that instead.
